core/arch/arm/plat-stm32mp1/main.c

```diff
@@ -188,8 +189,33 @@ static TEE_Result init_stm32mp1_drivers(void)
 
 	return TEE_SUCCESS;
 }
+
 service_init_late(init_stm32mp1_drivers);
 
+static TEE_Result init_late_stm32mp1_drivers(void)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+
+	/* Set access permission to TAM backup registers */
+	if (IS_ENABLED(CFG_STM32_TAMP)) {
+		struct stm32_bkpregs_conf conf = {
+			.nb_zone1_regs = TAMP_BKP_REGISTER_ZONE1_COUNT,
+			.nb_zone2_regs = TAMP_BKP_REGISTER_ZONE2_COUNT,
+		};
+
+		res = stm32_tamp_set_secure_bkpregs(&conf);
+		/* Driver is either already probed or is disabled */
```

Comment:
Why "or is disabled"? If it is disabled, `IS_ENABLED(CFG_STM32_TAMP)` is false isn't it?

Reply:
The driver may be embedded in optee but disabled in the DT (`status = '"disabled"`). 
Since this function is executed at `driver_init_late` initcall level, a return value of `TEE_ERROR_DEFER_INIT` means the driver was not found (from the DT) hence we consider it is disabled. In such case, we agree to discard TAMP registers secure hardening configuration.